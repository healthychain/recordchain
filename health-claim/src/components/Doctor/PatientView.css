.patient-view-container {
  display: flex;
  flex-direction: row;
}

.patient-img {
  width: 150px;
}
.patient-view-column {
  padding: 20px;
  justify-content: center;
  flex-direction: column;
  align-content: center;
  width: 50%;
}
