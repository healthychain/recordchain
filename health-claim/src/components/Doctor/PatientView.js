import React, { Component } from "react";
import PropTypes from "prop-types";
import "./PatientView.css";

class PatientView extends Component {
  componentDidMount() {
    const { id } = this.props;
    this.props.fetchClaims(id);
  }

  componentWillReceiveProps(newProps) {
    const { id } = this.props;
    const newId = newProps.id;

    if (newId !== id) {
      this.props.fetchClaims(newId);
    }
  }

  render = () => {
    return (
      <div className="patient-view-container">
        {this.props.loading ? (
          <p>Loading</p>
        ) : (
          <div>
            <div className="patient-view-column">
              {" "}
              <img
                alt="patient"
                className="patient-img"
                src="http://www.rw-designer.com/icon-image/14771-256x256x32.png"
              />
              <h4>Name: {this.props.name}</h4>
              <h4>ID: {this.props.id}</h4>
              <h4>
                Birthdate:
                {" 23. 4. 1977"}
              </h4>
            </div>
            <div className="patient-view-column">
              {this.props.claims.map(claim => (
                <p>{claim.title}</p>
              ))}
            </div>
          </div>
        )}
      </div>
    );
  };
}

PatientView.propTypes = {
  loading: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  birthDate: PropTypes.instanceOf(Date).isRequired,
  claims: PropTypes.array
};

PatientView.defaultProps = {
  claims: []
};

export default PatientView;
