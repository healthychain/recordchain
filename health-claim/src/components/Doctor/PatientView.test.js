import React from "react";
import renderer from "react-test-renderer";
import PatientView from "./PatientView";

describe("PatientView", () => {
  it("should render given name and ID", () => {
    const patient = renderer.create(
      <PatientView name="Name" id={10} fetchClaims={() => null} />
    );
    expect(patient).toMatchSnapshot();
  });
});
