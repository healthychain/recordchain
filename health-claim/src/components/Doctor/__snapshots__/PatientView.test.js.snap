// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientView should render given name and ID 1`] = `
<div
  className="patient-view-container"
>
  <div>
    <div
      className="patient-view-column"
    >
       
      <img
        alt="patient"
        className="patient-img"
        src="http://www.rw-designer.com/icon-image/14771-256x256x32.png"
      />
      <h4>
        Name: 
        Name
      </h4>
      <h4>
        ID: 
        10
      </h4>
      <h4>
        Birthdate:
         23. 4. 1977
      </h4>
    </div>
    <div
      className="patient-view-column"
    />
  </div>
</div>
`;
